src/daft-core/src/array/ops/list_agg.rs

```diff
@@ -10,52 +10,72 @@ use crate::{
     series::IntoSeries,
 };
 
+macro_rules! impl_daft_list_agg {
+    ($name:ty) => {
+        type Output = DaftResult<ListArray>;
+
+        fn list(&self) -> Self::Output {
+            let child_series = self.clone().into_series();
+            let offsets =
+                arrow2::offset::OffsetsBuffer::try_from(vec![0, child_series.len() as i64])?;
+            let list_field = self.field.to_list_field()?;
+            Ok(ListArray::new(list_field, child_series, offsets, None))
+        }
+
+        fn grouped_list(&self, groups: &GroupIndices) -> Self::Output {
```

Comment:
although not directly in scope of PR could really benefit from better docs on

https://github.com/Eventual-Inc/Daft/blob/d5b9a95a27fac14a916f3bbfffd8db0fdaaa8068/src/daft-core/src/array/ops/mod.rs#L200-L204

perhaps could include in the PR since related? I want to make sure I understand `grouped_list`

Reply:
I made a commit eacfe32b0c437c09ea51d333878db015dfece0e2

Reply:
I don't think this would be the best place to add comments, since this `[agg op]` and `grouped_[agg op]` pattern is pretty common across the various aggregations we do. Perhaps we could add some explanation about how they are used somewhere else.